app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+import com.hedvig.android.design.system.hedvig.RadioOptionDefaults.RadioOptionStyle.LeftAligned
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.LargeSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.MediumSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.SmallSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.LargeSizeRadioOptionTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.MediumSizeRadioOptionTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.SmallSizeRadioOptionTokens
+import hedvig.resources.R
+
+@Composable
+fun RadioGroup(
+  data: List<OptionData>,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+  groupLockedState: LockedState = NotLocked,
+  radioGroupStyle: RadioGroupStyle = RadioGroupDefaults.radioGroupStyle,
+  radioGroupSize: RadioGroupSize = RadioGroupDefaults.radioGroupSize,
+) {
+  val contentPadding = calculateContentPadding(radioGroupStyle, radioGroupSize)
+  when (radioGroupStyle) {
+    Horizontal -> HorizontalRadioGroup(
+      data = data,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is HorizontalWithLabel -> HorizontalRadioGroupWithLabel(
+      data = data,
+      groupLabelText = radioGroupStyle.groupLabelText,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is Vertical -> VerticalRadioGroup(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is VerticalWithGroupLabel -> VerticalRadioGroupWithLabel(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun HorizontalRadioGroup(
+  data: List<OptionData>,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  FlowRow(
+    modifier,
+    maxItemsInEachRow = 2,
+    overflow = Companion.Visible,
+    verticalArrangement = Arrangement.spacedBy(4.dp, Alignment.CenterVertically),
+  ) {
+    for (i in data) {
+      val itemPadding = if (data.indexOf(i) % 2 != 0) {
```

Comment:
Since we seem to do `indexOf()` every time inside the loop, and often more than once too, I'd suggest we turn this into a loop where we have the index already like this https://github.com/HedvigInsurance/android/blob/12a8a8a3c450bc1274782baa551fc3b5ff20d4b2/app/feature/feature-insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/InsuranceDestination.kt#L259
The IDE provides an auto-complete for this if you "option + enter" while having the cursor on `in` or perhaps `data` here I think.